### PR TITLE
[MTurk] Testing for the dev branch

### PR DESCRIPTION
**Pull request by JackUrb**
**Patch description**
As I destroyed many assumptions about the underlying structure that the MTurk unit and integration tests had made, it was time to revamp all of these tests to actually work again. This PR fixes all of these tests. In some cases it involved transitioning to new implementation details, and in many it involved removing tests that were covering weird corner cases that the new simpler infrastructure had removed (for a net change of -1014 lines!).

**Testing steps**
```
python test_db_interactions.py; python test_full_system.py; python test_mturk_agent.py; python test_mturk_manager.py; python test_socket_manager.py; python test_worker_manager.py
....
----------------------------------------------------------------------
Ran 4 tests in 0.042s
OK
......
----------------------------------------------------------------------
Ran 6 tests in 31.458s
OK
.........
----------------------------------------------------------------------
Ran 9 tests in 0.349s
OK
...........................
----------------------------------------------------------------------
Ran 27 tests in 58.393s
OK
..............
----------------------------------------------------------------------
Ran 14 tests in 53.030s
OK
.........
----------------------------------------------------------------------
Ran 9 tests in 0.042s
```
No test failures

### CODE REVIEW

File: parlai/mturk/core/dev/test/test_full_system.py [hunk 1]
```diff
@@ -415,35 +426,40 @@ def run_task_wait():
...
 
     def tearDown(self):
-        self.agent_1.always_beat = False
-        self.agent_2.always_beat = False
         for key in self.worlds_agents.keys():
             self.worlds_agents[key] = True
         self.mturk_manager.shutdown()
         self.fake_socket.close()
-        self.task_thread.join()
+        # if self.task_thread.isAlive():
```
Comment (klshuster) [inline parlai/mturk/core/dev/test/test_full_system.py]: nit: remove dead code

Comment (JackUrb) [reply parlai/mturk/core/dev/test/test_full_system.py]: Oops this should very much still be alive. Nice catch 👍
